features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/ThrottlingPolicies/Create/CreateBlackListPolicy.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom'
+
+import { Link } from 'react-router-dom';
+import AppBar from 'material-ui/AppBar';
+import Toolbar from 'material-ui/Toolbar';
+import IconButton from 'material-ui/IconButton';
+import Button from 'material-ui/Button';
+import MenuIcon from 'material-ui-icons/Menu';
+import Menu, { MenuItem } from 'material-ui/Menu';
+import Typography from 'material-ui/Typography';
+import Divider from 'material-ui/Divider';
+import Grid from 'material-ui/Grid';
+import Paper from 'material-ui/Paper';
+import TextField from 'material-ui/TextField'
+
+import GeneralDetails from '../Shared/GeneralDetails';
+
+import API from '../../../data/api';
+import Message from '../../Shared/Message';
+import '../Shared/Shared.css';
+import Alert from '../../Shared/Alert'
+import BlackListDetails from '../Shared/BlackListDetail';
+
+const messages = {
+    success: 'Created API rate limit successfully',
+    failure: 'Error while creating API rate limit',
+};
+
+class CreateBlackListPolicy extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            policy: {
+                conditionId: '',
+                conditionType: 'API',
+                conditionValue: '',
+                status: true,
+                ipCondition: {
+                    ipConditionType: '',
+                    specificIP: '',
+                    startingIP: '',
+                    endingIP: ''
+                }
+
+            },
+            value: 'API',
+        };
+        this.handleChangeChild = this.handleChangeChild.bind(this);
+        this.handleChangeChildValue = this.handleChangeChildValue.bind(this);
+    }
+
+
+    handleChangeChild(event) {
+        const policy = this.state.policy;
+        var value = event.target.value
+        policy.conditionType = value;
+        if (value === 'IP' || value === 'IP_RANGE') {
+            policy.ipCondition.ipConditionType = value;
+        }
+        this.setState({ value: event.target.value, policy: policy });
+    };
+
+    handleChangeChildValue(event) {
+        const policy = this.state.policy;
```

Comment:
Please use object destructuring 
```javascript
const { policy } = this.state;
```

Reply:
Advantage of this for single variable ?, when to use this and when not to. If we use it like this all the time we will have to use it like this